### DISCUSSION TITLE
Problems specifying a port for the needle trigger.

### DISCUSSION THREAD

**Discussion by gwcSoftware** [grblHAL/Plugin_embroidery#3] [Q&A]
I am confused as to how the port to be used for the needle trigger should be defined. According to the Readme, $452 should be used to specify the port and defaults to $452=2. I have the Openbuilds BlackboxX32 and was wanting to use the Safety Door connector as input for the needle trigger. It is defined to use GPIO_NUM_16 as AUXINPUT1_PIN in the BlackBoxX32_map.h file.

That option will not be used and is disabled by default. 

How can I define GPIO_NUM_16 to be used by the Embroidery Plugin for the needle trigger.

**Comment by gwcSoftware** [comment]
After looking at the BlackboxX32_map.h, I see that AUXINPUT1_PIN is defined as GPIO_NUM_16 but it is unclear as to why you specify the number 1 in the $452 setting. Even after setting $452=1 which should actually be GPIO16, if you run $pins, it still shows pin 22, (Aux 2) as being the embroidery trigger.

**Reply by terjeio** [reply]
Changing $452 returned an error, I have now commited a fix for that.
